### DISCUSSION TITLE
Steering motor won't Return to center

### DISCUSSION THREAD

**Discussion by connorthompson824** [pybricks/support#1992] [Q&A]
Hello! 

I just started using Pybricks today.  Coding through the LEGO Powered Up app is as far as my coding experience goes.  I followed the tutorial in the Racing Brick video on YouTube and here's my problem:

"RuntimeError: The steering mechanism has no end stop. Did you build a car yet?"

![pybricks_program_2024-12-26T20_04_21](https://github.com/user-attachments/assets/3968fb77-2fe4-4c25-960b-f5e8372c2a73)

I'm using a Technic Powered Up Large Motor in my build.  When I run my program, my motor just keeps going in one direction for a few seconds and then stops. Almost acting as if it's the drive motor but it's not. Why won't it act as a servo?

I'm humbly asking for your help.

Brett

**Comment by laurensvalk** [comment]
> When I run my program, my motor just keeps going in one direction for a few seconds and then stops.

The motor is looking for the end stop in your steering mechanism, which it needs to determine the center. If it doesn't find one in 10 seconds, it gives up.

Does it take that long to reach the end of your steering mechanism or... "Did you build a car yet?" 😁

**Reply by connorthompson824** [reply]
That makes sense. My gear rack for my car is brick built and is getting ripped apart each test. Is there a way to code the motor so that it will skip the test and just have its own set endpoints?

**Reply by BertLindeman** [reply]
Hi,

A similar challenge I had, which I solved by setting the maximum torque lower.

So less force is used to determine the left and right steering end points.

In the doc: `class Car(steer_motor, drive_motors, torque_limit=100)`
[See the doc page](https://code.pybricks.com/static/docs/v2.19.0/robotics.html?highlight=torque#pybricks.robotics.Car)  

But you use block coding and I do not know how to set the torque_limit with blocks.

You might try to copy the python code from your block program and paste it in a new python program.
(Remove the first comment line)

Change the definition of the car
from:
```python
car = Car(steer, drive)
```
to:
```python
car = Car(steer, drive, 20)  # set the torque_limit from 100% to a lower value
```

Just to see if this works for you.

Bert
